src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -49,11 +53,27 @@ uint32_t GetInterfaceId(chip::Inet::InterfaceId interfaceId)
     return interfaceId.IsPresent() ? interfaceId.GetPlatformInterface() : kDNSServiceInterfaceIndexAny;
 }
 
-std::string GetHostNameWithDomain(const char * hostname)
+std::string GetHostNameWithLocalDomain(const char * hostname)
 {
     return std::string(hostname) + '.' + kLocalDot;
 }
 
+bool HostNameHasDomain(const char * hostname, const char * domain)
+{
+    size_t domainLength   = strlen(domain);
+    size_t hostnameLength = strlen(hostname);
+    if (domainLength > hostnameLength)
+    {
+        return false;
+    }
+    const char * found = strstr(hostname, domain);
+    if (found == nullptr)
+    {
+        return false;
+    }
+    return (strncmp(found, domain, domainLength) == 0);
```

Comment:
Isn't this meant to check for a suffix? `strstr` will find a match anywhere within the string, and the `strncmp` will always succeed because the comparison is restricted to `domainLength`. Also, shouldn't the check also include the `.` before `domain`?

Reply:
will fix.